sycl/source/detail/memory_manager.cpp

```diff
@@ -176,27 +175,40 @@ void copyH2D(SYCLMemObjT *SYCLMemObj, char *SrcMem, QueueImplPtr SrcQueue,
   DstAccessRange[0] *= DstElemSize;
   SrcSize[0] *= SrcElemSize;
   DstSize[0] *= DstElemSize;
-
-  cl_command_queue CLQueue = UseExclusiveQueue
-                                 ? TgtQueue->getExclusiveQueueHandleRef()
-                                 : TgtQueue->getHandleRef();
-
+  
   if (1 == DimDst && 1 == DimSrc) {
     CHECK_OCL_CODE(clEnqueueWriteBuffer(
-        CLQueue, DstMem,
+        TgtQueue->getHandleRef(), DstMem,
         /*blocking_write=*/CL_FALSE, DstOffset[0], DstAccessRange[0],
         SrcMem + DstOffset[0], DepEvents.size(), &DepEvents[0], &OutEvent));
-  } else {
+    } else {
+
+    // test if the whole buffer is copied
+    if( SrcSize[0]   == SrcAccessRange[0] &&  SrcSize[1]   == SrcAccessRange[1]  &&  SrcSize[2]   == SrcAccessRange[2] &&
+	SrcSize[0]   == DstSize[0]        &&  SrcSize[1]   == DstSize[1]         &&  SrcSize[2]   == DstSize[2]        &&
+	SrcSize[0]   == DstAccessRange[0] &&  SrcSize[1]   == DstAccessRange[1]  &&  SrcSize[2]   == DstAccessRange[2] &&
+	SrcOffset[0] == DstOffset[0]      &&  SrcOffset[1] == DstOffset[1]       &&  SrcOffset[2] == DstOffset[2]      &&
+	SrcOffset[0] == 0                 &&  SrcOffset[1] == 0                  &&  SrcOffset[2] == 0 ) {
+      
+      CHECK_OCL_CODE(clEnqueueWriteBuffer(
+        TgtQueue->getHandleRef(), DstMem,
```

Comment:
I'd prefer we maintain the usage of the CLQueue variable unless there is good reason

Reply:
And also indentation similar to the original invocation of clEnqueueWriteBuffer, just to maintain uniform indentation